sdk/search/search-documents/test/node/searchIndexClient.spec.ts

```diff
@@ -0,0 +1,218 @@
+      assert.include(synonyms, synonymMap.synonyms[0]);
+      assert.include(synonyms, synonymMap.synonyms[1]);
+    });
+
+    it("throws error for invalid synonymmap object", async function(){
+      let retrievalError:boolean = false;
+      try {
+        await indexClient.getSynonymMap("garbxyz");
+      } catch(ex) {
+        retrievalError = true;
+      }
+      assert.isTrue(retrievalError);
+    });
+
+    it("creates the synonymmap object using createOrUpdateSynonymMap", async function(){
+      let synonymMap:SynonymMap = {
+        name: `my-azure-synonymmap-5`,
+        synonyms: ["United States, United States of America => USA", "Washington, Wash. => WA"]
+      };
+      await indexClient.createOrUpdateSynonymMap(synonymMap);
+      try {
+        synonymMap = await indexClient.getSynonymMap("my-azure-synonymmap-5");
+        assert.equal(synonymMap.name, "my-azure-synonymmap-5");
+        assert.equal(synonymMap.synonyms.length, 2);
+        const synonyms = ["United States, United States of America => USA", "Washington, Wash. => WA"];
+        assert.include(synonyms, synonymMap.synonyms[0]);
+        assert.include(synonyms, synonymMap.synonyms[1]);
+      } catch (ex) {
+        throw ex;
+      } finally {
+        await indexClient.deleteSynonymMap(synonymMap);
+      }
+    });
+
+    it("modify and updates the synonymmap object", async function(){
+      let synonymMap = await indexClient.getSynonymMap("my-azure-synonymmap-1");
+      synonymMap.synonyms.push("California, Clif. => CA");
+      await indexClient.createOrUpdateSynonymMap(synonymMap);
+      synonymMap = await indexClient.getSynonymMap("my-azure-synonymmap-1");
+      assert.equal(synonymMap.synonyms.length, 3);
+      const synonyms = ["United States, United States of America => USA", "Washington, Wash. => WA", "California, Clif. => CA"];
+      assert.include(synonyms, synonymMap.synonyms[0]);
+      assert.include(synonyms, synonymMap.synonyms[1]);
+      assert.include(synonyms, synonymMap.synonyms[2]);
+    });
+  });
+
+  describe("#indexes", function() {
+    it("gets the list of indexes", async function(){
+      const result = await indexClient.listIndexes();
+      let listOfIndexes = await result.next();
+      const indexNames:string[] = [];
+      while (!listOfIndexes.done) {
+        indexNames.push(listOfIndexes.value.name);
+        listOfIndexes = await result.next();
+      }
+      assert.include(indexNames, `hotel-live-test3`);
+    });
+
+    it("gets the list of indexes names", async function(){
+      const result = await indexClient.listIndexesNames();
+      let listOfIndexNames = await result.next();
+      const indexNames:string[] = [];
+      while (!listOfIndexNames.done) {
+        indexNames.push(listOfIndexNames.value);
+        listOfIndexNames = await result.next();
+      }
+      assert.include(indexNames, `hotel-live-test3`);
```

Comment:
seems like this should use `TEST_INDEX_NAME`

Reply:
Done